src/main/webapp/forum-script.js

```diff
@@ -0,0 +1,89 @@
+$( document ).ready( getForum );
+
+/**
+ *  Populates forum-placeholder with forum data
+ */
+function getForum() {
+  const id = -1;
+  const placeholder = $('#forum-placeholder');
+  expandForum(placeholder, id);
+}
+
+/**
+ *  Populates the replies for a forum element with the given id
+ *  in the placeholder given
+ *
+ *  @param {S.fn.init} placeholder the div that will hold the forum elements
+ *  @param {long} id the long that identifies the parent of the forum
+ *  elements
+ */
+function expandForum(placeholder, id) {
+  placeholder.empty();
+  fetch('/forum?id=' + id.toString())
+      .then((response) => (response.json())).then((elements) => {
+        for (let i = 0; i < elements.length; i++) {
+          createForumElement(placeholder, elements[i]);
+        }
+      });
+}
+
+/**
+ *  Creates a new forum element in the placeholder populated with
+ *  fields of the element given
+ *
+ *  @param {S.fn.init} placeholder the div that will hold the forum elements
+ *  @param {ForumElement} element the ForumElement that contains the data for
+ *  the element
+ */
+function createForumElement(placeholder, element) {
+  /* Creates a new div elements and adds to placeholder */
+  const elementDiv = $('<div></div>');
+  const elementId = 'element-' + element.id.toString();
+  elementDiv.attr('id', elementId);
+  placeholder.append(elementDiv);
+
+  /* Loads forumElement.html template and then populates div with the
+  fields from element */
+  elementDiv.load('forumElement.html', function() {
```

Comment:
This is really a question for Mike and myself, but it would be great if there's an easy way to use a light-weight template system with data binding to avoid hardcoding HTML.

In our app we use Angular but that's a huge framework to adopt.

Reply:
Yeah I wasnt quite sure how to handle this. Happy to implement a new system.

Reply:
Mustache may be a good place to start. It seems pretty lightweight, but I've never used it: https://mustache.github.io/mustache.5.html

There's plenty of usage examples out there too if you search for them.

Reply:
Resolving for now, I filed https://github.com/googleinterns/granite2020/issues/16 to investigate Mustache.